packages/venia-ui/lib/RootComponents/Category/category.gql.js

```diff
@@ -0,0 +1,58 @@
+import gql from 'graphql-tag';
+
+import { ProductDetailsFragment } from '../Product/product.gql';
+
+export const GET_CATEGORY_DATA = gql`
+    query category(
+        $id: Int!
+        $pageSize: Int!
+        $currentPage: Int!
+        $onServer: Boolean!
+        $filters: ProductAttributeFilterInput!
+        $sort: ProductAttributeSortInput
+    ) {
+        category(id: $id) {
+            id
+            description
+            name
+            product_count
+            meta_title @include(if: $onServer)
+            meta_keywords @include(if: $onServer)
+            meta_description
+        }
+        products(
+            pageSize: $pageSize
+            currentPage: $currentPage
+            filter: $filters
+            sort: $sort
+        ) {
+            items {
+                ...ProductDetailsFragment
+                # Even though these are already requested in the fragment,
+                # the server may return an incorrect response if you don't hard
+                # code the fields.
+                # https://github.com/magento/graphql-ce/issues/1027
+                description {
+                    html
+                }
+                media_gallery_entries {
+                    id
+                    label
+                    position
+                    disabled
+                    file
+                }
+                meta_description
+                small_image {
+                    url
+                }
+                url_key
```

Comment:
This sucks :/ I thought switching to a fragment would prevent the extra fetch that is happening in the PDP (because we pre-fetch?) but it's still occurring even if I remove the `cache-and-network` policy. Probably something to do with the noted bug? But it's been closed and the repo archived so for now we get a comment and some hard coded fields.

Reply:
Yeah, agreed - I was going to comment here too. Not sure why we get an extra fetch though, the whole idea of the PR that introduced all this (#1935) was that the category page would pre-fetch all the details the product pages needed.

Reply:
My concern is that the data fetched by the category query is not being "seen" by the product query when it checks the cache. This could be related to a misconfiguration or a bug in apollo client, so I'll investigate.

As far as the bug I referenced, it seems like it'll get fixed in 2.4.1 which is good but doesn't help us now :/

Reply:
After discussion with @supernova-at we realized that the original fragment reads were intentional because otherwise apollo client doesn't know how to match the PDP query and the category query. For example, when the category query prefetches products, it stores them in a cache entry keyed with the args to the query. When the PDP loads it checks for a different cache key because all we have at that time is the url_key. What results is a cache miss, so we make the network call.

We have options:

a) See if we can somehow get the category to prefetch a query that is keyed by url_key, so that the PDP will get a cache hit.

b) Remove the prefetch from the category page.

Both are fine, IMO, and I'll see if I can get `a` to work first before removing the category prefetching entirely.

Reply:
"a" did not work but I did move prefetching to the gallery item. May be overstepping the responsibility of the gallery item component tbh.

Reply:
Resolved through prefetching with IntersectionObserver in galleryItem